src/main/java/seedu/address/logic/parser/AddLeaveCommandParser.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE_END;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE_START;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TITLE;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AddLeaveCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.leave.Date;
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class AddLeaveCommandParser implements Parser<AddLeaveCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddLeaveCommand parse(String args) throws ParseException {
+        Index index;
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TITLE, PREFIX_DATE_START,
+                        PREFIX_DATE_END, PREFIX_DESCRIPTION);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_TITLE, PREFIX_DATE_START,
+                PREFIX_DATE_END)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddLeaveCommand.MESSAGE_USAGE));
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_TITLE, PREFIX_DATE_START,
+                PREFIX_DATE_END, PREFIX_DESCRIPTION);
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddLeaveCommand.MESSAGE_USAGE), pe);
+        }
+
+        String title = ParserUtil.parseTitle(argMultimap.getValue(PREFIX_TITLE).get());
+        Date dateStart = ParserUtil.parseDateStart(argMultimap.getValue(PREFIX_DATE_START).get());
+        Date dateEnd = ParserUtil.parseDateEnd(argMultimap.getValue(PREFIX_DATE_END).get(),
+                argMultimap.getValue(PREFIX_DATE_START).get());
+
+        String description = "none";
```

Comment:
“none” should be a constant, and it might be better to put it in a if-else statement so that it's clearer how the "none" state is obtained